qa/tasks/mgr/test_progress.py

```diff
@@ -149,6 +149,19 @@ def _osd_count(self):
         osd_map = self.mgr_cluster.mon_manager.get_osd_dump_json()
         return len(osd_map['osds'])
 
+    def _set_no_recovery_backfill(self, flag):
+        # Set recovery and backfill on or off
+        if flag:
+            self.mgr_cluster.mon_manager.raw_cluster_cmd(
+                'osd', 'set', 'norecover')
+            self.mgr_cluster.mon_manager.raw_cluster_cmd(
+                'osd', 'set', 'nobackfill')
+        else:
+            self.mgr_cluster.mon_manager.raw_cluster_cmd(
+                'osd', 'unset', 'norecover')
+            self.mgr_cluster.mon_manager.raw_cluster_cmd(
```

Comment:
i am not sure if `flag` is the best name. because normally, flag should be a number (not a boolean) indicating a status or a required status.

```suggestion
    def _set_no_recovery_backfill(self, do_set):
        # Set recovery and backfill on or off
        prefix = 'osd set' if do_set else 'osd unset'
        for key in ['nobackfill', 'norecover']:
            command = prefix.split() + [key]
            self.mgr_cluster.mon_manager.raw_cluster_cmd(
                *command)
```